[fea] friend status Enum 처리

J-ungry opened the pull request:
- status Enum 처리
- reject 상태 제거

[todo]
- jooq 도입
- 친구 신청 수락
- 친구 신청 거절
- 테스트 코드 작성

J-ungry committed:
[fea] friend status Enum 처리

- status Enum 처리
- reject 상태 제거

[todo]
- jooq 도입

devy1540 commented on src/main/java/com/example/toygry/moneymoa/Friends/service/FriendsService.java:
```diff
@@ -75,9 +76,9 @@ public List<UserListDto> getUserList(String token, String id) throws Exception {
         // 이미 친구인지 확인하기 친구라면 에러 뱉기
         switch (dto.status()) {
-            case "ACCEPT" -> throw new FriendDuplicateException();
+            case FriendsStatus.ACCEPT -> throw new FriendDuplicateException();
```
enum 나중에 static import로 빼면 좀 더 깔끔해져용

J-ungry committed:
[fea] friend status Enum 처리

- status Enum 처리
- reject 상태 제거

[todo]
- jooq 도입
- 친구 신청 수락
- 친구 신청 거절
- 테스트 코드 작성

devy1540 reviewed (approved):
lgtm
